flake.nix

```diff
@@ -121,6 +121,21 @@
             ''))}
             tar --sort=name --mtime='@1' --owner=0 --group=0 --numeric-owner -c tf-ncl/*.ncl | pixz -t > $out/tf-ncl.tar.xz
           '';
+
+          test-single-example = template: pkgs.writeShellScript "test-${template}" ''
+            set -e -x
+            temp_directory=$(mktemp -d)
+            trap 'rm -r -- "$temp_directory"' EXIT
```

Comment:
I'm admittedly not a great bash programmer. It's setting up the cleaning code for removing the temporary directory upon exit?